napari/components/viewer_model.py

```diff
@@ -997,6 +1006,7 @@ def open_sample(
         from napari.plugins import _npe2, plugin_manager
 
         plugin_spec_reader = None
+        data: Union[None, SampleDataCreator, SampleData]
```

Comment:
```suggestion
        data: None | SampleDataCreator | SampleData
```

Reply:
I'm not for using of `|` operator, even with `from __future__ import annotation`, this creates problem as `|` will not properly work with `typing.get_type_hints(...)`:

```python
from __future__ import annotations
from typing import get_type_hints

def f(a: int | float):
    pass

get_type_hints(f) # TypeError: unsupported operand type(s) for |: 'type' and 'type'
```

Reply:
does this problem will still exit in python 3.9?

Reply:
Yes.

Reply:
I guess it will as `|` is introduced in 3.10 and `get_type_hints` uses runtime.

Reply:
(code updated to reflect this). And it starts working with 3.10

`from __future__ import annotation` only change that annotation are stored as strings. It justs delay evaluation so in general it does not mean you can use more recent runtime features.